logstash-core/src/main/java/org/logstash/execution/ShutdownWatcherExt.java

```diff
@@ -172,7 +172,8 @@ public IRubyObject start(final ThreadContext context) throws InterruptedExceptio
                         if (stalledCount == 0) {
                             LOGGER.error("The shutdown process appears to be stalled due to busy or blocked plugins. Check the logs for more information.");
                             if (isPqDraining) {
-                                LOGGER.info("The queue is draining before shutdown.");
+                                String pipelineId = pipeline.callMethod(context, "pipeline_id").asJavaString();
+                                LOGGER.info("The queue for pipeline {} is draining before shutdown.", pipelineId);
```

Comment:
I wonder how easy it'd be to show the event count here as well. to let users know if there's progress or if it's stalled.

Reply:
we could print also the `stalledCount` to get, on the debug output, something like
```
The queue for pipeline my_pipeline is draining before shutdown (1/3).
```
since this PR has already been merged, I could take care of it in another, let me know, it's  an easy change